README
Dist::Zilla::App::Command::bakeini

"bakeini" is an "App::Command" module for "Dist::Zilla" that enables one
to have two versions of their "dist.ini", one which contains their
bundle, and the other which is generated from the first in a static and
portable way, without requiring the bundle to be present.

This allows contributors and test targets to have a mostly "static"
configuration that is less prone to randomly breaking your distributions
every time you change something significant in your bundle.

It also allows contributors to only need the dependencies they really
need, not the super-set of dependencies your bundle probably implies.

And at the same time, you still have the flexibility and power you
normally have with a centralized configuration stored in a bundle, which
you can roll out on demand, instead of having the roll out automatically
propagate every time the bundle gets updated.

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-App-Command-bakeini

0.002007 2017-03-04T06:05:08Z 457d08c

0.002006 2017-03-04T05:46:16Z 0c7ac74
 [BugFix]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-App-Command-bakeini/issues/3

 [Dependencies::Stats]
 - Dependencies changed since 0.002005, see misc/*.deps* for details
 - develop: +5 ↑2 -1 (suggests: ↑1 -1)
 - test: ↓1

0.002005 2015-04-12T06:49:58Z e518885
 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - configure: (recommends: ↓1)
 - develop: +3 ↑3 -1 (recommends: ↓1, suggests: ↑2)
 - runtime: (recommends: +1)
 - test: +3 -1 (recommends: ↓2)

 [Minor]
 - Tests rearchitected in dztest

0.002000 2014-11-08T15:20:00Z 4fbaf17
 [Dependencies::Stats]
 - Dependencies changed since 0.001004, see misc/*.deps* for details
 - configure: (recommends: ↑1)
 - develop: (recommends: ↑1, suggests: ↑1)
 - runtime: ↑1
 - test: (recommends: ↑2)

 [Features]
 - Now Supports --comments=all|authordeps|none ( default is all )
 - Thanks to Alan Young (AYOUNG), #1 and #2

0.001004 2014-10-30T23:29:28Z 6170f9a
 [Dependencies::Stats]
 - Dependencies changed since 0.001003, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +3 -1 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1
 - test: +1 (recommends: ↑2)

 [Internals]
 - stop requiring utf8 ( optimisation )
 - 5.006 is now minimum.

0.001003 2014-10-14T17:57:42Z 203f4a7
 [Dependencies::Stats]
 - Dependencies changed since 0.001002, see misc/*.deps* for details
 - develop: +3 ↑1 (suggests: ↑2)
 - test: (recommends: ↑1)

 [Metadata]
 - Set Author=kentnl@cpan.org

0.001002 2014-08-16T05:19:57Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - tests updated.
 - metadata updated.
 - whitespace reduced.

 [Dependencies::Stats]
 - Dependencies changed since 0.001001, see misc/*.deps* for details
 - configure: ↓1
 - develop: +2 ↑1 (suggests: ↑2)
 - test: +1 -3 (recommends: ↑1 -1)

 [Metadata]
 - move to kentnl org

 [Misc]
 - Improve source side POD to reduce build side whitespace.

 [Tests]
 - update Test::ReportPrereqs

0.001001 2014-07-12T15:21:03Z
 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - develop: -1 (suggests: +2 -1)
 - test: +2

 [Documentation]
 - Add tips and tricks section indicating how to make bakeini-specific plugins.

 [Tests]
 - Added basic application test.

0.001000 2014-06-03T02:49:50Z
 - First version.


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    ini
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

#$bundle->remove_policy('ErrorHandling::RequireCarping');
$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "bake dist.ini to not need the bundles.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-App-Command-bakeini",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::App::Command::bakeini",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::Util::ExpandINI" => "0.003000",
    "Path::Tiny" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Util::Test::KENTNL" => "1.005000",
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "List::Util" => 0,
    "Test::DZil" => 0,
    "Test::Differences" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.002007",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Dist::Zilla::App" => 0,
  "Dist::Zilla::Util::ExpandINI" => "0.003000",
  "Dist::Zilla::Util::Test::KENTNL" => "1.005000",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "List::Util" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Differences" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
